include/gunrock/util/performance.hxx

```diff
@@ -0,0 +1,183 @@
+#include <sys/utsname.h>
+#include "nlohmann/json.hpp"
+#include <time.h>
+#include <regex>
+#include <filesystem>
+#include <fstream>
+
+using vertex_t = int;
+using edge_t = int;
+
+class Sysinfo {
+ private:
+  struct utsname uts;
+
+ public:
+  Sysinfo() { uname(&uts); }
+
+  std::string sysname() const { return std::string(uts.sysname); }
+  std::string release() const { return std::string(uts.release); }
+  std::string version() const { return std::string(uts.version); }
+  std::string machine() const { return std::string(uts.machine); }
+  std::string nodename() const { return std::string(uts.nodename); }
+
+  void get_sys_info(nlohmann::json* jsn) {
+    std::map<std::string, std::string> sysinfo;
+    sysinfo["sysname"] = sysname();
+    sysinfo["release"] = release();
+    sysinfo["version"] = version();
+    sysinfo["machine"] = machine();
+    sysinfo["nodename"] = nodename();
+    jsn->push_back(nlohmann::json::object_t::value_type("sysinfo", sysinfo));
+  }
+};
+
+void get_gpu_info(nlohmann::json* jsn) {
```

Comment:
Can you potentially use this insead? https://github.com/gunrock/gunrock/blob/main/include/gunrock/cuda/device_properties.hxx

If the functionality doesn't exist (maybe you need a version of driver and its not in there), please move that from this bit of code in that file or relevant `cuda/*` file and just call it here. Let's talk on slack about this as well.